### PR TITLE
Backend refactoring

### LINKED ISSUE #1653: Refactor go backend
I think it would be nice to decouple backend code a bit. 

Especially `resource/pod/detail.go`, which shouldn't import daemon set, job etc. To do it I'd suggest making interface for it or using not resource list but specific structure. At the moment `Controller` struct seems to be a bit too big and forces us to use `detail` and `list` packages inside our resource packages.

### PR DESCRIPTION
**Pull request by maciaszczykm**
- [x] remove all `list` and `detail` packages from resources packages
- [x] in order to do so backend for getting pod creator has to be changed
- [x] and frontend has to be updated to display created by card corrrectly
- [x] zerostate
- [x] tests

Closes #1653.

### CODE REVIEW

File: src/app/backend/resource/pod/detail.go [hunk 1]
```diff
@@ -136,10 +114,8 @@ func GetPodDetail(client k8sClient.Interface, heapsterClient client.HeapsterClie
 		return nil, err
 	}
 
-	controller := Controller{
-		Kind: "unknown",
-	}
-	creatorAnnotation, found := pod.ObjectMeta.Annotations[api.CreatedByAnnotation]
+	controller := owner.ResourceOwner{}
+	creatorAnnotation, found := pod.ObjectMeta.Annotations[api.CreatedByAnnotation] // TODO Retwite to owned by
```
Comment (olekzabl) [inline src/app/backend/resource/pod/detail.go]: typo: "Retwite"

File: src/app/backend/resource/pod/detail.go [hunk 2]
```diff
@@ -183,18 +160,21 @@ func getPodCreator(client k8sClient.Interface, creatorAnnotation string, nsQuery
...
 	reference := serializedReference.Reference
 
-	controller, err := toPodController(client, reference, pods.Items, events.Items, heapsterClient)
+	rc, _ := owner.NewResourceController(reference, client)
+
+	controller := rc.Get(pods.Items)
+
+	fmt.Println(controller)
+
+	// TODO
```
Comment (olekzabl) [inline src/app/backend/resource/pod/detail.go]: I guess this TODO means "improve error handling", but I'm not sure (particularly as there's a blank line after it).
Can you specify its meaning?

Comment (maciaszczykm) [inline src/app/backend/resource/pod/detail.go]: @olekzabl Yeah, this part has to be finished. May I ping you when it will be ready for review?

Comment (olekzabl) [inline src/app/backend/resource/pod/detail.go]: Sure.

Comment (maciaszczykm) [inline src/app/backend/resource/pod/detail.go]: @olekzabl Okay, thank you :)

**Comment by maciaszczykm** [comment]
@olekzabl PTAL

File: src/app/frontend/poddetail/creatorinfo.html [hunk 3]
```diff
@@ -16,52 +16,114 @@
...
+                                        grow="nogrow">
+        </kd-resource-card-header-column>
+      </kd-resource-card-header-columns>
+      <kd-resource-card object-meta="$ctrl.creator.objectMeta"
+                        type-meta="$ctrl.creator.typeMeta">
+        <kd-resource-card-status layout="row">
+          <md-icon class="material-icons md-warn"
+                   ng-if="::$ctrl.hasWarnings()">
+            error
+            <md-tooltip>One or more pods have errors</md-tooltip>
```
Comment (olekzabl) [inline src/app/frontend/poddetail/creatorinfo.html]: Just making sure if "warnings" from the code correspond to "errors" for the user?

Comment (maciaszczykm) [inline src/app/frontend/poddetail/creatorinfo.html]: Yes, if one of the pods has some errors then controller status will be displayed as red error icon just like below:

image

**Comment by maciaszczykm** [comment]
Still one tiny issue to fix with closure comments.

**Comment by maciaszczykm** [comment]
@olekzabl Thanks!

**Pull request merged**